clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -3765,7 +3765,7 @@ ConstantAddress CodeGenModule::GetAddrOfTemplateParamObject(
   auto *GV = new llvm::GlobalVariable(getModule(), Init->getType(),
                                       /*isConstant=*/true, Linkage, Init, Name);
   setGVProperties(GV, TPO);
-  if (supportsCOMDAT())
+  if (supportsCOMDAT() && Linkage == llvm::GlobalValue::LinkOnceODRLinkage)
```

Comment:
I recall that in some cases WeakODRLinkage is used.  Should the condition be something like `!isLocalLinkage()`?

Reply:
Linkage is set to either LinkOnceODRLinkage or InternalLinkage a few lines above this.